be/src/olap/rowset/rowset_meta_manager.cpp

```diff
@@ -413,8 +413,9 @@ Status RowsetMetaManager::remove(OlapMeta* meta, TabletUid tablet_uid, const Row
 }
 
 Status RowsetMetaManager::remove_binlog(OlapMeta* meta, const std::string& suffix) {
-    return meta->remove(META_COLUMN_FAMILY_INDEX,
-                        {kBinlogMetaPrefix.data() + suffix, kBinlogDataPrefix.data() + suffix});
+    std::vector<std::string> keys = {kBinlogMetaPrefix.data() + suffix,
```

Comment:
I test by gcc12. It can be compiled kindly.

```bash
./build_one.sh rowset_meta_manager.cpp src/olap/CMakeFiles/Olap.dir/rowset/rowset_meta_manager.cpp.o
[1/1] ccache Doris/doris_toolchain/bin/g++ -DBOOST_STACKTRACE_USE_BACKTRACE -DHAVE_INTTYPES_H -DHAVE_NETINET_IN_H -DUSE_HADOOP_HDFS....

Doris/doris_toolchain/bin/g++ --version
g++ (Ubuntu 12.3.0-1ubuntu1~22.04) 12.3.0
Copyright (C) 2022 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```